src/libsystemd/sd-journal/journal-def.h

```diff
@@ -57,7 +57,6 @@ struct ObjectHeader {
         uint8_t flags;
         uint8_t reserved[6];
         le64_t size;
-        uint8_t payload[];
```

Comment:
Hmm, this one seems a bit moot and the GNU extension actually makes sense IMO, since it clearly indicates where the data lies.

Reply:
But, that's mostly unused.

Reply:
why not just keep this with [0] at least? it would alleviate our concerns but also address the issue at hand?